# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/darling-dmg/hotfix-gcc16.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Fix build with GCC 16: add the RecordIterator decrement operator used by
std::advance, and include <functional> for std::function.

--- a/src/HFSBTreeNode.h
+++ b/src/HFSBTreeNode.h
@@ -129,6 +129,11 @@ public:
 			m_index++;
 			return *this;
 		}
+		RecordIterator& operator--()
+		{
+			m_index--;
+			return *this;
+		}
 
 		difference_type operator-(const RecordIterator& that)
 		{
--- a/src/main-fuse.cpp
+++ b/src/main-fuse.cpp
@@ -1,4 +1,5 @@
 #include "main-fuse.h"
+#include <functional>
 #include <cstring>
 #include <iostream>
 #include <cstdio>
